### PR TITLE
BLOCKDATA redesign

### PR DESCRIPTION
**Pull request by OlivierBBB**
Redesign of the `BLOCKDATA` module
- delegates computations to the WCP and EUC modules
- several new checks
- allows for `BLOCKGASLIMIT` to change in accordance with the yellow paper
- compatible with the conflation containing the Genesis block

### CODE REVIEW

File: block_data/binarities.tex [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+		\item $\previousConflation$
+		\item $\currentConflation$
+		\item $\isCoinbase$
+		\item $\isTimestamp$
+		\item $\isNumber$
+		\item $\isDifficulty$
+		\item $\isGaslimit$
+		\item $\isChainid$
+	\end{enumerate}
+\end{multicols}
```
Comment (lorenzogentile404) [inline block_data/binarities.tex:14]: `IS_BASEFEE` is missing.

File: block_data/shorthands.tex [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+		\previousConflation & \longleftrightarrow & \locIsPrevConf   \\
+		\currentConflation  & \longleftrightarrow & \locIsCurrConf   \\
+		\isCoinbase         & \longleftrightarrow & \locIsCoinbase   \\
+		\isTimestamp        & \longleftrightarrow & \locIsTimestamp  \\
+		\isNumber           & \longleftrightarrow & \locIsNumber     \\
+		\isDifficulty       & \longleftrightarrow & \locIsDifficulty \\
+		\isGaslimit         & \longleftrightarrow & \locIsGaslimit   \\
+		\isChainid          & \longleftrightarrow & \locIsChainid    \\
+		\isBasefee          & \longleftrightarrow & \locIsBasefee    \\
+	\end{array} \right.
```
Comment (lorenzogentile404) [inline block_data/shorthands.tex:13]: I suggest to defined those as aliases of columns names using: https://www.notion.so/consensys/Naming-Conventions-55b293749c9a459eaef5c96d7f982267?pvs=4#91deab67ed894ae78b1311beb8380443.

File: block_data/binarities.tex (hunk 1 above)
Comment (OlivierBBB) [reply block_data/binarities.tex:15]: Indeed, fixed.

File: block_data/computations/coinbase.tex [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+\begin{description}
+	\item[\underline{\underline{Setting \inst{COINBASE}:}}]
+		we impose
+		\[
+			\left\{ \begin{array}{lcl}
+				\blockDataHi _{i} & = & \coinbase\high _{i} \\
+				\blockDataLo _{i} & = & \coinbase\low  _{i} \\
+			\end{array} \right.
+		\]
+	\item[\underline{\underline{\inst{GASLIMIT} deviation lower bound:}}]
```
Comment (amkCha) [inline block_data/computations/coinbase.tex:18]: Should it be : 
`\item[\underline{\underline{\inst{COINBASE} lower bound:}}]`

**Pull request merged**